File: Day48_SumRan.cpp
#include <bits/stdc++.h>
using namespace std;
ifstream in("in.txt");
ofstream out("out.txt");

vector<string> summaryRanges(vector<int> &nums)
{
    vector<string> ans;
    int n = nums.size();
    if (!n)
        return ans;
    int l = nums[0], r = nums[0];
    for (int i = 1; i < n; i++)
    {
        r = nums[i];
        if ((nums[i] - nums[i - 1]) > 1)
        {
            if (l == nums[i - 1])
                ans.push_back(to_string(l));
            else
                ans.push_back(to_string(l) + "->" + to_string(nums[i - 1]));
            l = r;
        }
    }
    if (l == nums[n - 1])
        ans.push_back(to_string(l));
    else
        ans.push_back(to_string(l) + "->" + to_string(nums[n - 1]));
    return ans;
}

int main()
{

    return 0;
}

File: Day48_SumRan.py
def summaryRanges(nums: list[int]) -> list[str]:
    if(not len(nums)):
        return []
    ans = []
    l = nums[0]
    r = nums[0]
    for i in range(1, len(nums)):
        r = nums[i]
        if(nums[i] - nums[i-1] > 1):
            if(l == nums[i-1]):
                ans.append(str(l))
            else:
                ans.append(str(l) + '->' + str(nums[i-1]))
            l = r
    if(l == nums[-1]):
        ans.append(str(l))
    else:
        ans.append(str(l) + '->' + str(nums[-1]))
    return ans

print(summaryRanges([0,1,2,4,5,7]))